Kubernetes scheduler unit test registry_test Failures with master go

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


#### What happened:
Below are the failures while running unit test **pkg/scheduler/framework/runtime/registry_test.go** with golang in master branch.
```
--- FAIL: TestMerge (0.00s)
    --- FAIL: TestMerge/valid_Merge (0.00s)
        registry_test.go:158: Merge(). Expected map[pluginFactory1:0x11607ea0 pluginFactory2:0x11607ed0]. Got map[pluginFactory1:0x11607e40 pluginFactory2:0x11607e70] instead.
    --- FAIL: TestMerge/Merge_duplicate_factories (0.00s)
        registry_test.go:158: Merge(). Expected map[pluginFactory1:0x11607f60]. Got map[pluginFactory1:0x11607f00] instead.
--- FAIL: TestRegister (0.00s)
    --- FAIL: TestRegister/valid_Register (0.00s)
        registry_test.go:207: Register(). Expected map[pluginFactory1:0x11607fc0]. Got map[pluginFactory1:0x11607f90] instead.
    --- FAIL: TestRegister/Register_duplicate_factories (0.00s)
        registry_test.go:207: Register(). Expected map[pluginFactory1:0x11608050]. Got map[pluginFactory1:0x11607ff0] instead.
--- FAIL: TestUnregister (0.00s)
    --- FAIL: TestUnregister/valid_Unregister (0.00s)
        registry_test.go:252: Unregister(). Expected map[pluginFactory2:0x116080e0]. Got map[pluginFactory2:0x116080b0] instead.
```
**Reference : https://github.com/golang/go/issues/45697#issuecomment-826007480**

**$ go version
go version devel go1.17-5daefc5363 Thu Apr 22 01:40:02 2021 +0000 linux/ppc64le**
Note: Occurs in x86 architecture as well.

#### What you expected to happen:
The tests should Pass as with stable go
```
[root@rajalakshmi1-workspace kubernetes]# go version
go version go1.16 linux/ppc64le
[root@rajalakshmi1-workspace kubernetes]# make test KUBE_RACE=-race WHAT=./pkg/scheduler/framework/runtime/ 2>&1 | tee pss.log
+++ [0422 09:55:38] Running tests without code coverage
ok      k8s.io/kubernetes/pkg/scheduler/framework/runtime       0.190s
[root@rajalakshmi1-workspace kubernetes]#
```

#### How to reproduce it (as minimally and precisely as possible):
git clone https://github.com/kubernetes/kubernetes.git
cd kubernetes
make test KUBE_RACE=-race WHAT=./pkg/scheduler/framework/runtime/

#### Environment:
Master branch of kubernetes repo.
https://github.com/kubernetes/kubernetes
OS: Linux
Observed failure on ppc64le and x86 architectures

**Below is the attached log:**
[fail.log](https://github.com/kubernetes/kubernetes/files/6404969/fail.log)

--- a/pkg/scheduler/framework/runtime/registry_test.go
+++ b/pkg/scheduler/framework/runtime/registry_test.go
@@ -76,7 +76,7 @@ func TestDecodeInto(t *testing.T) {
 func isRegistryEqual(registryX, registryY Registry) bool {
 	for name, pluginFactory := range registryY {
 		if val, ok := registryX[name]; ok {
-			if reflect.ValueOf(pluginFactory) != reflect.ValueOf(val) {
+			if reflect.ValueOf(pluginFactory).Pointer() != reflect.ValueOf(val).Pointer() {
 				// pluginFactory functions are not the same.
 				return false
 			}

